runtime/lua/vim/lsp/inlay_hint.lua

```diff
@@ -43,17 +43,11 @@ function M.on_inlayhint(err, result, ctx, _)
     return
   end
   local bufnr = assert(ctx.bufnr)
-  if util.buf_versions[bufnr] ~= ctx.version then
+  if util.buf_versions[bufnr] ~= ctx.version or not result or not bufstates[bufnr].enabled then
```

Comment:
why not check `nvim_buf_is_loaded` here?

Reply:
bufstates[bufnr] has removed in on_detach  so check before this function invoke . reduce a function call instruction

Reply:
Honestly, it works the same everywhere. this lsp handler just calls this function. your suggest is more uniform. I just wonder if there will be an extra call assembly instruction 🥲

Reply:
I wouldn't worry about performance here, because this condition is unusual. So readability is more important, and keeping the "return" conditions in one place seems very helpful.